### PR TITLE
JS changes to work on non-root URLs

### PR DESCRIPTION
**Pull request by srs81**
Hi, thanks for the great plugin. I found one problem where it doesn't work on non-root URLs. 

I see that you solved the problem using the window.location and Javascript, but for some reason, it doesn't work on my app install.

I've re-factored the code slightly to get the "firstLevel" variable from CakePHP itself, so that way its always getting the right URL. This also reduces the Javascript dependency for getting the path information.

### CODE REVIEW

File: View/Reports/index.ctp [hunk 1]
```diff
@@ -22,6 +22,12 @@ OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
 WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 */
 ?>
+<script type="text/javascript">
+	firstLevel = "<?php echo Router::url('/') ?>";
```
Comment (luisdias) [inline View/Reports/index.ctp:26]: The Router::url('/') does not return an ending slash ? 

Without ; at the end of the line

File: webroot/js/index.js [hunk 2]
```diff
@@ -22,14 +22,6 @@
...
-            var pathArray = pathName.split("/");
-            var firstLevel = null;
-            if ( pathName.substr(0,1) == "/" )       
-                firstLevel = "/" + pathArray[1];
-            else
-                firstLevel = "/" + pathArray[0];        
-
             $.ajax({
             type: "POST",
             url: firstLevel+"/report_manager/reports/ajaxGetOneToManyOptions",
```
Comment (luisdias) [inline webroot/js/index.js:27]: Because of the Router::url('/') ending slash :

firstLevel+"report_manager/reports/ajaxGetOneToManyOptions",

**Comment by srs81** [comment]
I think it does for me - it doesn't give the trailing slash on your end?

**Comment by luisdias** [comment]
My point is: if there is an ending slash, you do not need the initial
slash on the index.js file at the url ( right before report_manager :
firstlevel + '/report_manager ...) otherwise we will have double slash
in the firstlevel variable.

What do you think?

By the way, thank you for your cooperation!

2012/5/2 Suman reply@reply.github.com:

> I think it does for me - it doesn't give the trailing slash on your end?
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/luisdias/CakePHP-Report-Manager-Plugin/pull/1#issuecomment-5452995

## 

# 

# L.D.

**Comment by srs81** [comment]
Ah yes, totally agreed. :) Sorry about that. I interpreted your question the opposite way - somehow I thought the Router::url('/") was _not_ returning a slash. The additional / is just in case, but not necessary, like you say.

**Pull request merged**